### ISSUE TITLE
TypeVar argument "cannot be assigned" to a matching input type

### ISSUE THREAD

**Issue by danlkv** [microsoft/pyright#1771]
**Describe the bug**
I'm writing a simple function to merge an iterable of sets:

```python
from functools import reduce
from typing import Iterable, TypeVar

T = TypeVar('T', set, frozenset)
def merge_sets(sets: Iterable[T]) -> T:
    return functools.reduce(lambda x, y: x.union(y), sets)
```
at the argument to `.union(y)` pyright gives an error:
```
            pyright  [Pyright reportGeneralTypeIssues] [E] Argument of type "T@merge_sets" cannot
            be assigned to parameter "s" of type "Iterable[Unknown]" in function "union"
            Type "T@merge_sets" cannot be assigned to type "Iterable[Unknown]"
```
But both `set` or `frozenset` are perfectly iterable. In fact, if I change the T to either `set` or `frozenset` the error goes away.

**VS Code extension or command-line**

I'm running pyright with vim and coc-pyright, python 3.8

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.133, which I just published. It will also be in the next release of pylance.
